scripts/exglobal_stage_ic.sh

```diff
@@ -8,99 +8,110 @@ GDATE=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
+  for ((ii = 0; ii < "${NMEM_ENS}"; ii++)); do
+    MEMDIR_ARRAY+=("mem$(printf "%03d" "${ii}")")
+  done
+else
+  MEMDIR_ARRAY+=("")
+fi
+
 # Initialize return code
 err=0
 
-error_message(){
-    echo "FATAL ERROR: Unable to copy ${1} to ${2} (Error code ${3})"
+error_message() {
+  echo "FATAL ERROR: Unable to copy ${1} to ${2} (Error code ${3})"
 }
 
 ###############################################################
-# Start staging
-
-# Stage the FV3 initial conditions to ROTDIR (cold start)
-YMD=${PDY} HH=${cyc} generate_com -r COM_ATMOS_INPUT
-[[ ! -d "${COM_ATMOS_INPUT}" ]] && mkdir -p "${COM_ATMOS_INPUT}"
-source="${BASE_CPLIC}/${CPL_ATMIC}/${PDY}${cyc}/${CDUMP}/${CASE}/INPUT/gfs_ctrl.nc"
-target="${COM_ATMOS_INPUT}/gfs_ctrl.nc"
-${NCP} "${source}" "${target}"
-rc=$?
-(( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
-err=$((err + rc))
-for ftype in gfs_data sfc_data; do
-  for tt in $(seq 1 6); do
-    source="${BASE_CPLIC}/${CPL_ATMIC}/${PDY}${cyc}/${CDUMP}/${CASE}/INPUT/${ftype}.tile${tt}.nc"
-    target="${COM_ATMOS_INPUT}/${ftype}.tile${tt}.nc"
-    ${NCP} "${source}" "${target}"
-    rc=$?
-    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
-    err=$((err + rc))
-  done
-done
-
-# Stage ocean initial conditions to ROTDIR (warm start)
-if [[ "${DO_OCN:-}" = "YES" ]]; then
-  YMD=${gPDY} HH=${gcyc} generate_com -r COM_OCEAN_RESTART
-  [[ ! -d "${COM_OCEAN_RESTART}" ]] && mkdir -p "${COM_OCEAN_RESTART}"
-  source="${BASE_CPLIC}/${CPL_OCNIC}/${PDY}${cyc}/ocn/${OCNRES}/MOM.res.nc"
-  target="${COM_OCEAN_RESTART}/${PDY}.${cyc}0000.MOM.res.nc"
-  ${NCP} "${source}" "${target}"
+for MEMDIR in "${MEMDIR_ARRAY[@]}"; do
+  # Stage the FV3 initial conditions to ROTDIR (cold start)
+  YMD=${PDY} HH=${cyc} generate_com -r COM_ATMOS_INPUT
+  [[ ! -d "${COM_ATMOS_INPUT}" ]] && mkdir -p "${COM_ATMOS_INPUT}"
+  src="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${MEMDIR}/atmos/gfs_ctrl.nc"
+  tgt="${COM_ATMOS_INPUT}/gfs_ctrl.nc"
+  ${NCP} "${src}" "${tgt}"
   rc=$?
-  (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
+  ((rc != 0)) && error_message "${src}" "${tgt}" "${rc}"
   err=$((err + rc))
-  case "${OCNRES}" in
-    "500" | "100")  # Only 5 degree or 1 degree ocean does not have MOM.res_[1-4].nc files
-    ;;
-    "025")  # Only 1/4 degree ocean has MOM.res_[1-4].nc files
-      for nn in $(seq 1 4); do
-        source="${BASE_CPLIC}/${CPL_OCNIC}/${PDY}${cyc}/ocn/${OCNRES}/MOM.res_${nn}.nc"
-        if [[ -f "${source}" ]]; then
-          target="${COM_OCEAN_RESTART}/${PDY}.${cyc}0000.MOM.res_${nn}.nc"
-          ${NCP} "${source}" "${target}"
-          rc=$?
-          (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
-          err=$((err + rc))
-        fi
-      done
-    ;;
-    *)
-      echo "FATAL ERROR: Unsupported ocean resolution ${OCNRES}"
-      rc=1
+  for ftype in gfs_data sfc_data; do
+    for tt in $(seq 1 6); do
```

Comment:
```suggestion
    for ((tt = 1; tt <= 6; tt++)); do
```
To be consistent w/ the other loops in this script.